share/config.py

```diff
@@ -264,3 +272,107 @@ def parse_config(config_yaml: str) -> Config:
         conf.add_input(current_input)
 
     return conf
+
+
+def aws_sm_expander(config_yaml: str) -> str:
+    needed_secret_arns = {}
+    needed_secret_keys = {}
+
+    re_pattern = r"arn:aws:secretsmanager:(?:[^:]*):(?:[^:]*):secret:(?:[^\n]*)"
+    found_secrets = re.findall(re_pattern, config_yaml)
+
+    for secret in found_secrets:
+        secret = secret.strip('"').strip("'")
+        splitted_secret = secret.split(":")
+
+        if len(splitted_secret) != 7 and len(splitted_secret) != 8:
+            raise SyntaxError("Invalid arn format: {}".format(secret))
+
+        if secret not in needed_secret_arns:
+            needed_secret_arns[secret] = ""
+
+        region = splitted_secret[3]
+        secrets_manager_name = splitted_secret[6]
+
+        if region == "":
+            raise ValueError("Must be provided region in arn: {}".format(secret))
+
+        if secrets_manager_name == "":
+            raise ValueError("Must be provided secrets manager name in arn: {}".format(secret))
+
+        if region not in needed_secret_keys:
+            needed_secret_keys[region] = {}
+
+        if secrets_manager_name not in needed_secret_keys[region]:
+            needed_secret_keys[region][secrets_manager_name] = ""
+
+    for region in needed_secret_keys:
+        for secrets_manager_name in needed_secret_keys[region]:
+            str_secrets = get_secret_values(secrets_manager_name, region)
+            parsed_secrets = parse_secrets_str(str_secrets)
+
+            needed_secret_keys[region][secrets_manager_name] = parsed_secrets
+
+    for secret in needed_secret_arns:
```

Comment:
nit: `secret` has the same name here as before with a different content, let's consider a suffix to clarify what's the actual content here and avoid confusion